web/web-content.py

```diff
@@ -0,0 +1,187 @@
+        return contents
+    return directory_dict
+
+
+def generate_page(env, template_name, file_name, destination, headline, content):
+    """
+    Create a specific html page of the open205.github.io website from a Jinja2 template.
+    """
+    template = env.get_template(template_name)
+    html_file = os.path.join(destination, file_name)
+    with open(html_file, 'w+') as f:
+        f.write(template.render(
+            nav = file_name,
+            headline = headline,
+            content = content
+        ))
+    f.close()
+
+
+def create_files(web_dir):
+    """
+    Create or replace the example and template files inside of open205.github.io
+    """
+    assets_dir = os.path.join(web_dir, 'assets')
+    if not os.path.isdir(assets_dir):
+        os.mkdir(assets_dir)
+
+    examples_dir = os.path.join(assets_dir, 'examples')
+    if not os.path.isdir(examples_dir):
+        os.mkdir(examples_dir)
+
+    json_dir = os.path.join(examples_dir, 'json')
+    if not os.path.isdir(json_dir):
+        os.mkdir(json_dir)
+
+    cbor_dir = os.path.join(examples_dir, 'cbor')
+    if not os.path.isdir(cbor_dir):
+        os.mkdir(cbor_dir)
+
+    xlsx_dir = os.path.join(examples_dir, 'xlsx')
+    if not os.path.isdir(xlsx_dir):
+        os.mkdir(xlsx_dir)
+
+    yaml_dir = os.path.join(examples_dir, 'yaml')
+    if not os.path.isdir(yaml_dir):
+        os.mkdir(yaml_dir)
+
+    templates_dir = os.path.join(assets_dir, 'templates')
+    if not os.path.isdir(templates_dir):
+        os.mkdir(templates_dir)
+
+    schema_dir = os.path.join(assets_dir, 'schema')
+    if not os.path.isdir(schema_dir):
+        os.mkdir(schema_dir)
+
+    tk205.translate_directory('schema-205/examples/json', json_dir)
+    tk205.translate_directory('schema-205/examples/json', cbor_dir)
+    tk205.translate_directory('schema-205/examples/json', xlsx_dir)
+    tk205.translate_directory('schema-205/examples/json', yaml_dir)
+    copy_tree('schema-205/schema', schema_dir)
+
+    # xlsx_template_creation
+    tk205.file_io.clear_directory(templates_dir)
+    template_content = tk205.load(os.path.join(root_dir, "..", "config", "templates.json"))
+    for template in template_content:
+        file_name_components = [template["RS"]]
+        if template["file-name-suffix"]:
+            file_name_components.append(template["file-name-suffix"])
+        file_name_components.append("template.a205.xlsx")
+        file_name = '-'.join(file_name_components)
+        tk205.template(template["RS"],os.path.join(templates_dir,file_name), **template["keywords"])
+
+
+def clone():
+    """
+    Clone open205.github.io website repository into the build directory.
+    """
+    build_dir = os.path.join(root_dir, '..', 'build')
+    if not os.path.isdir(build_dir):
+        os.mkdir(build_dir)
+    web_dir = os.path.join(build_dir, 'web')
+    if not os.path.isdir(web_dir):
+        git.Repo.clone_from("https://github.com/open205/open205.github.io.git", web_dir, branch='master')
```

Comment:
Interesting question. I felt that it was OK to assume that they had git since that is the most likely way you would acquire toolkit205, but this is probably and error we want to handle. From the gitpython docs: "GitPython needs the git executable to be installed on the system and available in your PATH for most operations." So not only do they need git, but it needs to be available on the path, which is usually not the case for windows users (but I think this is guaranteed to work if they are trying it from an anaconda environment).

The only thoughts I have about how we could handle this elegantly is to 
1. Wrap git commands in try/catch blocks. gitpython has a number of [exceptions](https://gitpython.readthedocs.io/en/stable/reference.html#git.exc.GitError) we could attempt to catch and exit the program.
2. Use python to try and call a simple git command from a shell and check the output.

Reply:
My latest commit wraps the entire cloning operation to handle a number of different situations regarding their build/web folder. This should allow them to work offline, continue on if they already have a repository, ect.